app/metal-controller-manager/api/v1alpha1/server_types.go

```diff
@@ -89,16 +89,25 @@ type ServerSpec struct {
 	ManagementAPI     *ManagementAPI          `json:"managementApi,omitempty"`
 	ConfigPatches     []ConfigPatches         `json:"configPatches,omitempty"`
 	Accepted          bool                    `json:"accepted"`
+	PXEBootAlways     bool                    `json:"pxeBootAlways,omitempty"`
 }
 
-// ConditionPowerCycle is used to control the powercycle flow.
-const ConditionPowerCycle clusterv1.ConditionType = "PowerCycle"
+const (
+	// ConditionPowerCycle is used to control the powercycle flow.
+	ConditionPowerCycle clusterv1.ConditionType = "PowerCycle"
+	// ConditionPXEBooted is used to record the fact that server got PXE booted.
+	ConditionPXEBooted clusterv1.ConditionType = "PXEBooted"
+)
 
 // ServerStatus defines the observed state of Server.
 type ServerStatus struct {
-	Ready      bool                  `json:"ready"`
-	InUse      bool                  `json:"inUse"`
-	IsClean    bool                  `json:"isClean"`
+	// +optional
```

Comment:
this is crucial to get correct `patch` operations working on the `Status`, as otherwise it fails with `.status.ready field being required`.